pkg/targets/adapter/googlecloudstoragetarget/adapter.go

```diff
@@ -79,15 +82,14 @@ func (a *googlecloudstorageAdapter) Start(ctx context.Context) error {
 }
 
 func (a *googlecloudstorageAdapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
-	switch typ := event.Type(); typ {
-	case v1alpha1.EventTypeGoogleCloudStorageObjectInsert:
+	if event.Type() == v1alpha1.EventTypeGoogleCloudStorageObjectInsert || a.discardCEContext {
+		return a.insertObjectData(ctx, event)
```

Comment:
We probably don't need two separate functions, do we?
We could pass just the desired data as []byte and I'd assume the rest of the logic to be the same in both cases.

Reply:
the reason we had two functions was to allow either arbitrary events to be passed in. or to allow the user to specify the filename and data explicitly via the "com.google.cloud.storage.object.insert" event.. I prefer the original implementation..